evmap/src/write.rs

```diff
@@ -705,8 +493,41 @@ where
                     entry.insert(Values::with_capacity_and_hasher(additional, hasher));
                 }
             },
+            Operation::MarkReady => {
+                inner.ready = true;
+            }
+            Operation::SetMeta(m) => {
+                inner.meta = m;
+            }
+            Operation::JustCloneRHandle => {
+                // This is applying the operation to the original read handle,
+                // which is empty, and needs to copy over all data from the
+                // write handle that we wrote to directly.
+
+                // XXX: it really is too bad that we can't just .clone() the data here and save
+                // ourselves a lot of re-hashing, re-bucketization, etc.
+                inner.data.extend(other.data.iter().map(|(k, vs)| {
+                    (
+                        k.clone(),
+                        Values::from_iter(
+                            vs.iter()
+                                .map(|v| unsafe { ManuallyDrop::into_inner((&*v).shallow_copy()) }),
+                            other.data.hasher(),
+                        ),
+                    )
+                }));
+            }
         }
     }
+
+    fn drop_first(self: Box<Self>) {
+        // Make sure that we don't drop values since we're only dropping the first shallow copy of
+        // each value.
+        //
+        // Safety: ManuallyDrop<T> has the same layout as T.
+        let inner: Box<Inner<K, ManuallyDrop<V>, M, S>> = unsafe { mem::transmute(self) };
```

Comment:
```suggestion
        let inner = unsafe { Box::from_raw(Box::into_raw(self) as *mut Inner<K, ManuallyDrop<V>, M, S>) };
```
Same general idea as the cast suggested above but with [Box](https://doc.rust-lang.org/std/boxed/struct.Box.html#method.into_raw).